src/config.h

```diff
@@ -40,8 +40,12 @@
 #include <fcntl.h>
 #endif
 
+#if defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
```

Comment:
May I know, What does `__MAC_OS_X_VERSION_MAX_ALLOWED` translates to?

Who is responsible for setting this MAX version value?

Is it Compiler?

Or Is it us? Do we define this via some Build file?

Reply:
`__MAC_OS_X_VERSION_MAX_ALLOWED` is to determine the highest macos version we want to run on.

it is determined by the SDK:
```c
#ifdef __MAC_OS_X_VERSION_MIN_REQUIRED
    /* make sure a default max version is set */
    #ifndef __MAC_OS_X_VERSION_MAX_ALLOWED
        #define __MAC_OS_X_VERSION_MAX_ALLOWED __MAC_13_3
    #endif
#endif /* __MAC_OS_X_VERSION_MIN_REQUIRED */
```

also refer to the document (Conditionally Compiling for Different SDKs): https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/cross_development/Using/using.html

Reply:
Thanks!